testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py

```diff
@@ -3896,6 +3897,115 @@ def test_blob_contructor_replication(params_from_base_test_setup, blob_data_type
+    [generate_2_png_100_100, True]
+])
+def test_replication_pull_from_empty_database(params_from_base_test_setup, attachment_generator, cbl_action_before_init_replication):
+    '''
+    @summary:
+    This test to validate pull replication on empty database. It should avoid sending deleted(tombstoned) documents.
+    1. create 100 docs in SGW
+    2. delete 10 docs in SGW, verify the 10 docs are in tombstone state
+    3. create a cbl db, but do not create any doc (cbl db is empty)
+    4. create a pull replicator, start replication
+    5. verify SGW stats, cbl_replication_pull.rev_send_count sent to CBL equals to 90
+    6. verify CBL, total number of docs is 90, and verify the doc ids are not belongs to tombstone docs
+    7. delete 15 more docs on SGW
+    8. verify SGW stats, cbl_replication_pull.rev_send_count sent to CBL is still (90 + 15 deleted) 105
+    9. verify CBL deleted docs in step 7 have been removed from CBL db
+    '''
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    sg_url = params_from_base_test_setup["sg_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    db = params_from_base_test_setup["db"]
+    db_config = params_from_base_test_setup["db_config"]
+
+    # Reset cluster to ensure no data in system
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    # 1. create 100 docs on SGW
+    sg_db = "db"
+    num_sg_docs = 100
+    channels = ["ABC"]
+    username = "autotest"
+    password = "password"
+
+    sg_client = MobileRestClient()
+    sg_client.create_user(sg_admin_url, sg_db, username, password=password, channels=channels)
+    cookie, session_id = sg_client.create_session(sg_admin_url, sg_db, username)
+    auth_session = cookie, session_id
+    sg_added_docs = sg_client.add_docs(url=sg_url, db=sg_db, number=num_sg_docs, id_prefix="sg_doc", channels=channels, auth=auth_session, attachments_generator=attachment_generator)
+
+    # 2a. delete 10 docs in SGW
+    docs_to_delete_1 = sg_added_docs[0:10]
+    sg_client.delete_docs(url=sg_url, db=sg_db, docs=docs_to_delete_1, auth=auth_session)
+    # 2b. verify the 10 deleted docs are in tombstone state
+    sg_docs = sg_client.get_all_docs(url=sg_admin_url, db=sg_db, include_docs=True)["rows"]
+    assert len(sg_docs) == 90, "Expected number of docs is incorrect after doc deletion in sync-gateway"
+    for sg_doc in sg_docs:
+        assert sg_doc not in docs_to_delete_1, "Deleted doc should not show on sg doc list"
+
+    # 3a. create a cbl db, but do not create any doc (cbl db is empty)
+    cbl_db_name = "empty-db-" + str(time.time())
+    cbl_db = db.create(cbl_db_name, db_config)
+    # 3b. alternate scenario - create docs on cbl then delete all docs on cbl
+    if cbl_action_before_init_replication:
```

Comment:
not sure why this condition required as I do not see steps in summary. Can you add it in summary? Also according to summary , we should not create docs in CBL and it supposed to empty? Will it test empty cbl case if this is added?

Reply:
fixed the summary, made the text describe what's doing in the code